hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -380,7 +380,7 @@ private OmMetadataManagerImpl(OzoneConfiguration conf, File dir, String name)
         checkSnapshotDirExist(checkpoint);
       }
       setStore(loadDB(conf, metaDir, dbName, false,
-          java.util.Optional.of(Boolean.TRUE), false, false));
+          java.util.Optional.of(Boolean.TRUE), Optional.of(maxOpenFiles), false, false));
```

Comment:
I don't think optional is good idea to use for primitives e.g. `maxOpenFiles`. Someone can  pass `maxOpenFiles` as zero or negative. Would it work as expected in the case?

I think you should check that it is non-zero positive number [here](https://github.com/apache/ozone/pull/4884/files#diff-d4e4dbf0e95db46eded21d40936d1b18cd738dc19fa10ec45dc88211b4b498e3R562) or in the `setMaxNumberOfOpenFiles`.

Reply:
`-1` should be a valid input IMO as that is the RocksDB default.

Reply:
I don't agree with overriding default with default. But it is OK if we want to allow that.

Zero is still no handled properly.